src/main/java/com/aws/greengrass/deployment/converter/DeploymentDocumentConverter.java

```diff
@@ -217,6 +217,8 @@ private static DeploymentPackageConfiguration convertComponent(String componentN
                         convertComponentUpdateOperation(componentUpdate.getConfigurationUpdate()));
         builder = builder.runWith(RunWith.builder()
                 .posixUser(componentUpdate.getRunWith() == null ? null : componentUpdate.getRunWith().getPosixUser())
```

Comment:
should add a comment since I didn't have one when I made the fix. Explain this interesting difference.

```
// We always want to set the RunWith even if the passed in RunWith is null in order to allow for either keeping the
// existing runwith user, updating it, or reverting it to the default.
```